spec/log_search_spec.rb

```diff
@@ -36,44 +36,65 @@
 
     it 'does search for two terms on the same line' do
       expect(subject.search('TestLogs/test.txt', And.new(
-                                                            Keyword.new('Utilities'),
-                                                            Keyword.new('ReadRegistry'))).length).to eq(7)
+        described_class.new('Utilities'),
+        described_class.new('ReadRegistry'))).length).to eq(7)
     end
 
     it 'does search for either term on the same line' do
       expect(subject.search('TestLogs/test.txt', Or.new(
-                                                            Keyword.new('SmartQueue'),
-                                                            Keyword.new('QueueOfMessages'))).length).to eq(3)
+        described_class.new('SmartQueue'),
+        described_class.new('QueueOfMessages'))).length).to eq(3)
     end
 
     it 'does search for any term from list on the line' do
       expect(subject.search('TestLogs/test.txt',Any.new(
-                                                            [Keyword.new('SmartQueue'),
-                                                             Keyword.new('QueueOfMessages')])).length).to eq(3)
+        [described_class.new('SmartQueue'),
+         described_class.new('QueueOfMessages')])).length).to eq(3)
     end
 
     it 'does search for all terms in a list on the line' do
       expect(subject.search('TestLogs/test.txt',All.new(
-                                                   [Keyword.new('Start'),
-                                                    Keyword.new('New Call'),
-                                                    Keyword.new('Interaction Created')])).length).to eq(5)
+        [described_class.new('Start'),
+         described_class.new('New Call'),
+         described_class.new('Interaction Created')])).length).to eq(5)
     end
 
     it 'does a complicated search' do
       expect(subject.search('TestLogs/test.txt',
-                                     Or.new(
-                                         And.new(
-                                             Keyword.new('INFO'),
-                                             Keyword.new('Utilities')),
-                                         And.new(
-                                             Keyword.new('INFO'),
-                                             Keyword.new('CVHTCall')))).length).to eq(14)
+                            Or.new(
+                              And.new(
+                                described_class.new('INFO'),
+                                described_class.new('Utilities')),
+                              And.new(
+                                described_class.new('INFO'),
+                                described_class.new('CVHTCall')))).length).to eq(14)
+    end
+
+    it 'should return an array containing the correct number of lines' do
+      expect(subject.search('TestLogs/test.txt', described_class.new('uTILITIES',mode: :insensitive)).length).to eq(11)
```

Comment:
Note that this syntax for setting the `mode` parameter works for *both* `Keyword`, which uses an options hash, and `KeywordPartDeux`, which uses keyword arguments. Notice that this code uses a colon `:` and not a hashrocket `=>`. This colon syntax was implemented in Ruby 1.9.x. Although the options hash approach within the method itself is backward compatible to Ruby 1.8.x, the calling syntax *here* only works on Ruby 1.9 and above.

Also note that this syntax assumes *symbols* as hash keys, not strings. Which is OK most of the time because symbols are vastly more efficient as hash keys than strings.